src/QSymbolicsBase/literal_objects.jl

```diff
@@ -92,4 +106,4 @@ symbollabel(x::SZero) =  "𝟎"
 basis(x::SZero) = nothing
 
 Base.show(io::IO, x::SZero) = print(io, symbollabel(x))
-Base.iszero(x::SymQObj) = isa(x, SZero)
+Base.iszero(x::Union{SymQObj, Symbolic{Number}, Symbolic{Complex}}) = isa(x, SZero)
```

Comment:
I think the new `iszero` implementations here are type piracy. Where are they needed?

Reply:
This fixes a bug mentioned in #54 as well: https://github.com/QuantumSavory/QuantumSymbolics.jl/issues/54#issuecomment-2187334722.